data-loader/core/src/main/java/com/scalar/db/dataloader/core/dataimport/dao/ScalarDBDao.java

```diff
@@ -0,0 +1,433 @@
+  public Scanner createScanner(
+      String namespace,
+      String table,
+      List<String> projectionColumns,
+      int limit,
+      DistributedStorage storage)
+      throws ScalarDBDaoException {
+    Scan scan =
+        createScan(namespace, table, null, null, new ArrayList<>(), projectionColumns, limit);
+    try {
+      return storage.scan(scan);
+    } catch (ExecutionException e) {
+      throw new ScalarDBDaoException(
+          CoreError.DATA_LOADER_ERROR_SCAN.buildMessage(e.getMessage()), e);
+    }
+  }
+
+  /**
+   * Create a ScalarDB scanner instance
+   *
+   * @param namespace ScalarDB namespace
+   * @param table ScalarDB table name
+   * @param partitionKey Partition key used in ScalarDB scan
+   * @param scanRange Optional range to set ScalarDB scan start and end values
+   * @param sortOrders Optional scan clustering key sorting values
+   * @param projectionColumns List of column projection to use during scan
+   * @param limit Scan limit value
+   * @param storage Distributed storage for ScalarDB connection that is running in storage mode
+   * @return ScalarDB Scanner object
+   * @throws ScalarDBDaoException if scan fails
+   */
+  public Scanner createScanner(
+      String namespace,
+      String table,
+      Key partitionKey,
+      ScanRange scanRange,
+      List<Scan.Ordering> sortOrders,
+      List<String> projectionColumns,
+      int limit,
+      DistributedStorage storage)
+      throws ScalarDBDaoException {
+    Scan scan =
+        createScan(namespace, table, partitionKey, scanRange, sortOrders, projectionColumns, limit);
+    try {
+      return storage.scan(scan);
+    } catch (ExecutionException e) {
+      throw new ScalarDBDaoException(
+          CoreError.DATA_LOADER_ERROR_SCAN.buildMessage(e.getMessage()), e);
+    }
+  }
+
+  /**
+   * Create ScalarDB scan instance
+   *
+   * @param namespace ScalarDB namespace
+   * @param table ScalarDB table name
+   * @param partitionKey Partition key used in ScalarDB scan
+   * @param scanRange Optional range to set ScalarDB scan start and end values
+   * @param sortOrders Optional scan clustering key sorting values
+   * @param projectionColumns List of column projection to use during scan
+   * @param limit Scan limit value
+   * @return ScalarDB scan instance
+   */
+  Scan createScan(
+      String namespace,
+      String table,
+      Key partitionKey,
+      ScanRange scanRange,
+      List<Scan.Ordering> sortOrders,
+      List<String> projectionColumns,
+      int limit) {
+    // If no partition key is provided a scan all is created
+    if (partitionKey == null) {
+      ScanBuilder.BuildableScanAll buildableScanAll =
+          Scan.newBuilder().namespace(namespace).table(table).all();
+
+      // projection columns
+      if (projectionColumns != null && !projectionColumns.isEmpty()) {
+        buildableScanAll.projections(projectionColumns);
+      }
+      // Can ordering be added?
+      for (Scan.Ordering sort : sortOrders) {
```

Comment:
It seems that `List<Scan.Ordering> projectionColumns` can be null but `List<String> sortOrders` cannot. 
It might be better to stay consistent with `Collection` type parameter, either parameter of type `List<...>` can be null or cannot.